lib/Cron/Workers/SubscriberLinkTokens.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace MailPoet\Cron\Workers;
+
+use MailPoet\Cron\CronHelper;
+use MailPoet\Models\Subscriber;
+use MailPoet\Models\ScheduledTask;
+
+if (!defined('ABSPATH')) exit;
+
+class SubscriberLinkTokens extends SimpleWorker {
+  const TASK_TYPE = 'subscriber_link_tokens';
+  const BATCH_SIZE = 1000;
+
+  function processTaskStrategy(ScheduledTask $task) {
+    $count = $this->addTokens();
+    while ($count === self::BATCH_SIZE) {
+      CronHelper::enforceExecutionLimit($this->timer);
+      $count = $this->addTokens();
+    };
+    if ($count > 0) {
+      self::schedule();
+    }
+    return true;
+  }
+
+  public function addTokens() {
+    $instances = Subscriber::whereNull('link_token')->limit(self::BATCH_SIZE)->findMany();
+    foreach ($instances as $instance) {
+      $instance->set('link_token', Subscriber::generateToken($instance->email));
+      $instance->save();
+    }
+    return count($instances);
+  }
+}
```

Comment:
One thing I am not so sure is the schedule. You left the default, which means it will run once per week. On the one hand, it is good because once it finishes, it will never do anything again, so it is good it is not running very often. But on the other hand, running once a week means it will have less time to do its job in the beginning. For bigger customers, it could take several months to finish. Is it a problem? I think not, but I am asking.

Reply:
I didn't think about this problem. In would override `getNextRunDate` to schedule immediately and keep the `schedule` call in `processTaskStrategy` so that it's only called if there are subscribers left. I would then override the `process` method to not run schedule at all. That way, once all subscribers are handled, no more task is scheduled. What do you think?

Reply:
I think that is a good idea.

Reply:
Done.

Reply:
Sorry, Amine, I have one more comment to this. Right now this implementation would cause this to run all the time non-stop.